logic/src/commonMain/kotlin/com/wire/kalium/logic/CoreFailure.kt

```diff
@@ -56,25 +56,19 @@ sealed class StorageFailure : CoreFailure() {
     class Generic(val rootCause: Throwable) : StorageFailure()
 }
 
-internal inline fun <T : Any> wrapApiRequest(networkCall: () -> NetworkResponse<T>): Either<NetworkFailure, T> {
-    return try {
-        when (val result = networkCall()) {
-            is NetworkResponse.Success -> Either.Right(result.value)
-            is NetworkResponse.Error -> {
-                kaliumLogger.e(result.kException.stackTraceToString())
-                val exception = result.kException
-                if (exception is KaliumException.GenericError && exception.cause is IOException) {
-                    Either.Left(NetworkFailure.NoNetworkConnection(exception))
-                } else {
-                    Either.Left(NetworkFailure.ServerMiscommunication(result.kException))
-                }
+internal inline fun <T : Any> wrapApiRequest(networkCall: () -> NetworkResponse<T>): Either<NetworkFailure, T> =
+    when (val result = networkCall()) {
+        is NetworkResponse.Success -> Either.Right(result.value)
+        is NetworkResponse.Error -> {
+            kaliumLogger.e(result.kException.stackTraceToString())
+            val exception = result.kException
+            if (exception is KaliumException.GenericError && exception.cause is IOException) {
+                Either.Left(NetworkFailure.NoNetworkConnection(exception))
+            } else {
+                Either.Left(NetworkFailure.ServerMiscommunication(result.kException))
             }
         }
-    } catch (e: Exception) {
-        kaliumLogger.e(e.stackTraceToString())
-        Either.Left(NetworkFailure.ServerMiscommunication(e))
     }
```

Comment:
We don't need to catch exceptions here.

The network layer already handles it. Any exception made by whoever is calling this should be handled by them. We should not incourage easy and generic exception handling.